packages/components/src/public/ExtendedMap.ts

```diff
@@ -0,0 +1,145 @@
+type KeyType = string | number | symbol;
```

Comment:
is number allowed? won't they be converted to string?

Reply:
Number is allowed, they will not be converted to strings, here is what happens in the JS console:

```TypeScript
const map = new Map();
map.set(1, "I have a number for a key");
console.log(typeof Array.from(map.keys())[0]);
```

That will output "number".